rfcs/0052-create-sink-into-table.md

```diff
@@ -0,0 +1,399 @@
+FROM TUMBLE(dwd_orders, order_time, INTERVAL '1' HOUR)
+WHERE window_start >= '2010-10-10 00:00:00'
+GROUP BY window_start, product
+WITH (
+  force_append_only = 'true'
+);
+```
+
+## Syntax, semantics and behaviors
+
+### create sink into table
+
+```SQL
+  CREATE SINK sink_name INTO table_name [( column_name [, ...])] AS select_query
+```
+All the behavior is the same with a sink into an external table with a connector. The number of the column_names should be the same with the number of the columns of the sink's schema. If the column names are not specified, the tables full schema(all not hidden columns).
+
+For the table with pk, all the pk columns should be determined the sink's schema. And for the table without pk, the sink must be append-only.
+
+A circular streaming plan is forbidden, with `dependent_relations` we can easily do it.
+
+### create table as 
+
+Additionally, we may introduce a `CREATE TABLE AS` syntax, which can be considered as a sugar of `CREATE TABLE (...)` and `CREATE SINK INTO` combined together. This syntax is also adopted by [Flink SQL](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=199541185) and [ksqlDB](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/create-table-as-select/).
+
+```sql
+CREATE TABLE table_name AS select_query
+```
+
+### alter mv to table
+
+To make it more usable for users, I would like to introduce another function to alter an exiting materialzied view to a table.
+
+```sql
+ALTER MATERIALIZED VIEW mv_table TO TABLE
+```
+
+This is because now we still recommend `create materialized view` at the first place instead of `create table as`. However, application developers are facing changes all the time, so it's very likely that they may realize they need to 'alter' an existing materialized view. Meanwhile, the strong consistency gurantee of MV prevents us to do any altering. With the help of `ALTER MATERIALIZED VIEW TO TABLE`, users can choose to break this consistency gurantee and alter their streaming pipelines.
+
+Currently, we can just support it with limitation that the mv's primary key can not be hidden column, we will discuss why in the "implementation" phrase.
+
+The statement requires **Allow a sink and a table have the same name**. It is ok in our system because the sink will never appear in the FROM clause. An alternative is that we can add a `_sink` suffix on the sink's name.
+
+### Implementation
+
+Thanks for @BugenZhao and @shanicky's offline discussion.
+#### CREATE SINK into TABLE
+
+Currently, the table's plan is shown as the graph. In brief, the table's data can comes from the external source or BatchDML. And the DMLExec is used here as a side input. ![](./images/0052-sink-to-table/cur-table-plan.png).
+
+The `create sink to table` introduces more data sources for the table. And now RW has implemented stream Union executor and RowIdGen with specified distribution. So we can support it more clearly with these new features.
```

Comment:
> stream Union executor

IIUC, you mean merging the events from multiple upstream into one stream? If so, please call it MergeExecutor or whatever. It's much different from `Union` in relational algebra.

Reply:
Yes, but as a streaming executor they are same 🤔

Reply:
Not exactly. It's more like a continuous version of Chain operator. Optimization stuff like PK derivation, selectivity estimation, etc. also differ, although the optimizer won't process it for now.

My proposal is that we can introduce a `Merge` operator in Optimizer side only, and both `Merge` and `Union All` will be converted to the existing `UnionExecutor` (which might be renamed to avoid ambiguity with `Union` in SQL)